From 5c41fe18806435f97366f120c7e858f71900f18a Mon Sep 17 00:00:00 2001
From: Bastien Nocera <hadess@hadess.net>
Date: Fri, 1 Jul 2022 13:32:42 +0200
Subject: [PATCH] backends: Fix traceback in artsearch plugin

Traceback (most recent call last):
  File "/app/lib/rhythmbox/plugins/artsearch/embedded.py", line 46, in discovered_cb
    pixbuf = RB.gst_process_embedded_image(tags, tagname)
  File "/usr/lib/python3.9/site-packages/gi/module.py", line 123, in __getattr__
    raise AttributeError("%r object has no attribute %r" % (
AttributeError: 'gi.repository.RB' object has no attribute 'gst_process_embedded_image'

Make sure to process rb-player-gst-helper.h as it used to be in
the autotools days, so we have a binding for
gst_process_embedded_image():
https://gitlab.gnome.org/GNOME/rhythmbox/-/blob/9b7c55544f0f6453fc6b764135cd01ba82168558/bindings/gi/Makefile.am#L37

Closes: #1992
---
 backends/meson.build | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/backends/meson.build b/backends/meson.build
index 3391ff939..e21c1d2f5 100644
--- a/backends/meson.build
+++ b/backends/meson.build
@@ -2,7 +2,9 @@ backends_headers = files(
   'rb-encoder.h',
   'rb-player.h',
   'rb-player-gst-filter.h',
-  'rb-player-gst-tee.h'
+  'rb-player-gst-tee.h',
+
+  'gstreamer/rb-player-gst-helper.h',
 )
 
 install_headers(backends_headers, subdir: 'rhythmbox/backends')
-- 
2.36.1

